src/cryptonote_core/tx_pool.cpp

```diff
@@ -1151,6 +1188,125 @@ namespace cryptonote
+
+              if (!second_pass)
+              {
+                LOG_PRINT_L2("Will check this tx again in the second pass");
+                ++nofake_txs_in_pool;
+                sorted_it++;
+                continue;
+              }
+
+              LOG_PRINT_L2("We have currently " << nofake_txs_taken << " non-private transactions out of " << n << " transactions in the block");
+
+              bool can_nofake_tx_be_simply_added = false;
+              if (nofake_txs_taken + 1 <= rational_ceil((n + 1) * NOFAKE_TXS_TO_TOTAL_TXS_PERCENT, 100))
+              {
+                LOG_PRINT_L2("Checking if this tx can be simply added");
+                uint64_t block_reward;
+                if(get_block_reward(median_size, total_size + meta.blob_size, already_generated_coins, block_reward, version, height))
+                {
+                  coinbase = block_reward + fee + meta.fee;
+                  if (coinbase >= template_accept_threshold(best_coinbase))
+                  {
+                    LOG_PRINT_L2("  adding this tx would increase coinbase to " << print_money(coinbase));
+                    can_nofake_tx_be_simply_added = true;
+                  }
+                }
+              }
+
+              bool can_nofake_tx_replace_existing_tx = false;
+              if (!can_nofake_tx_be_simply_added && n > nofake_txs_taken && nofake_txs_taken + 1 <= rational_ceil(n * NOFAKE_TXS_TO_TOTAL_TXS_PERCENT, 100))
+              {
+                LOG_PRINT_L2("Checking if this tx can replace an already added tx");
+                uint64_t block_reward;
+                if(get_block_reward(median_size, total_size + meta.blob_size - tx_sizes.back(), already_generated_coins, block_reward, version, height))
+                {
+                  coinbase = block_reward + fee + meta.fee - tx_fees.back();
+                  if (coinbase >= template_accept_threshold(best_coinbase))
+                  {
+                    LOG_PRINT_L2("  replacing existing tx " << bl.tx_hashes.back() << " with this tx would increase coinbase to " << print_money(coinbase));
+                    can_nofake_tx_replace_existing_tx = true;
+                    bl.tx_hashes.pop_back();
+                    tx_sizes.pop_back();
+                    tx_fees.pop_back();
+                  }
+                }
+              }
+
+              if (!can_nofake_tx_be_simply_added && !can_nofake_tx_replace_existing_tx)
+              {
+                LOG_PRINT_L2("Adding this tx would decrease coinbase");
+                sorted_it++;
+                continue;
+              }
+            }
+          }
+        }
+        else
+        {
+          CHECKED_GET_SPECIFIC_VARIANT(tx.vin[0], const txin_to_key, itk, false);
+          // discourage < 3-way-mix transactions by mining them only as the first tx in an empty block
+          if (n > 0 && itk.key_offsets.size() < 3)
```

Comment:
This is (and needs to be) exactly the same as the v1 rule of Aeon which is copied from https://github.com/aeonix/aeon/pull/11/files#diff-e0629185a2adeca3fbc0f3ce27a79d65R426.

This will be irrelevant after the fork.